module/VuFind/src/VuFind/Db/Service/UserServiceInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Database service interface for users.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+
+namespace VuFind\Db\Service;
+
+use VuFind\Db\Entity\UserEntityInterface;
+
+/**
+ * Database service interface for users.
+ *
+ * @category VuFind
+ * @package  Database
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:database_gateways Wiki
+ */
+interface UserServiceInterface extends DbServiceInterface
+{
+    /**
+     * Retrieve a user object from the database based on ID.
+     *
+     * @param string $id ID.
+     *
+     * @return UserEntityInterface
+     */
+    public function getById($id);
```

Comment:
The [existing Doctrine code in #2233](https://github.com/vufind-org/vufind/pull/2233/files#diff-90567344401af857617a13d4fecf78891a1d71fde8e7a01292603fb73b91716fR90) calls this getUserById. It probably makes sense to align these methods. Since there may not always be a perfect 1:1 alignment between classes and tables, I think specifying the type of entity being retrieved in the method name is probably helpful.